### PR TITLE
update to 1.0.0-dev0

### PR DESCRIPTION
**Pull request by zhindes**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidaqmx-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Updates to version 1.0.0-dev0.

### Why should this Pull Request be merged?

Allows us to publish pre-release versions to PyPI for further testing.

### What testing has been done?

* Built documentation and looked at it - looks good
* Ran `pytest` - no unexpected failures

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -57,10 +57,10 @@
 # built documents.
 #
 # The short X.Y version.
-version = u'0.9'
+version = u'1.0'
 # The full version. Since we adhere to semver, there is no reason to make this
 # more specific.
-release = u'0.9'
+release = u'1.0'
```
Comment (bkeryan) [inline docs/conf.py:63]: FYI, it's possible to get conf.py to pull this from pyproject.toml:

https://github.com/ni/measurementlink-python/blob/main/_docs_source/conf.py

This is outside the scope of this PR.

**Review by bkeryan — APPROVED** [review]

File: docs/conf.py (hunk 1 above)
Comment (zhindes) [reply docs/conf.py:63]: ooh nice, I'll play with that after this!
